All features implemented

anauroranon opened the pull request:
## Changelog

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

**Image Histogram:** 
Feature that allows the user to choose an image from the list and see its corresponding RBG histogram.

Added files: 
- _histograms.py:_ handles the user request to generate an histogram relative to the selected image.
- _histogram.js:_  JS script that generates the histogram corresponding to the selected image.
- _histogram_output_queue.html:_ output page containing the histogram
- _histogram_select.html:_ form template that allows the user to choose an image from a limited list of images.
- _histogram_form.py:_ form code that manages the data corresponding to the user's choice.

Modified files:
- _base.html_: added the link to image histogram page into the navbar


**Color Jitter**:
Feature that allows the user to set parameters to change brightness, contrast, saturation and hue of an image and then classify it.

Added files:
- _classifications_jitter.py_: A new route inspired by classifications.py that adds color jitter transformation to the selected image. It takes informations from the form and uses them to modify the image through the function modify_image.
- _color_jitter_form.py_: A form that inherits from classifications_form and adds new fields to allow the user to insert values for the color jitter.
- _classification_jitter_select.html_: A new html file for the Color-jitter form

Modified files:
- _base.html_: added the link to color jitter page into the navbar
- _classification_output_queue.html_: Modified back button to refresh and go back to the previous page


**Download button**
Features that allows the user to download the classification result in png format or json format

Added files:
- _json_results.py_: handle that triggers the download of the json file

Modified files:
- _classification_output_queue.html_: added download buttons and some css code
- _poll.js_: linked one button to the download of the classification chart 


**Upload Image**
Feature that allows the user to upload an image and classify it

Added files:
- _classifications_upload.py_: route that allows to upload an image and classify it
- _classification_upload.html_: this html file incudes the form that allows to upload a file

Modified files:
- _base.html_: added the link to upload image file into the navbar

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
It does not introduce any breaking change, everything is compatible with original code


* **Other information**:(does the pr fix some issues? Tag them with #<issue id>)
  Fixes #1, #2, #3, #4

ManuMassi committed:
Added:
- json_results.py : new route that handles the request to download the json file with the results
Modified:
- classification_output_queue.html : added the two buttons for downloading, and linked the first one to json_results.py

idduperezza committed:
Added files:

idduperezza committed:
Added files:
- histogram.js -> (not working yet) should create the histogram in the frontend
- histogram_form -> form for choosing the image
- histogram_output_queue -> output
- histogram_select
- histograms

Modified the base.html: it's now possible to choose between classification and image histograms

ManuMassi committed:
Modified:
- json_results.py : added indent to output file

idduperezza committed:
Modified files:
- histogram.js -> the script now calculates the image histogram based on RGB colors of the selected image and prints it in the web page.
- hisogram_output_queue.html -> added histogram title, and CSS style for the histogram
- histograms.py -> cleaned up the code

ManuMassi committed:
Modified:
- poll.js : implemented API to download the chart in png format
- classification_output_queue.html : linked the button to the API

anauroranon committed:
Added button for color jitter in navbar. Created Color Jitter Api and Color Jitter form.

anauroranon committed:
Added button for color jitter in navbar. Created Color Jitter Api and Color Jitter form.

anauroranon committed:
Adjusted color jitter transformation. Optimized forms modularity.

anauroranon committed:
Adjusted back button. Modified css for color jitter form.

anauroranon committed:
Added error messages

anauroranon committed:
minor changes

anauroranon committed:
style changes

anauroranon committed:
Style changes

anauroranon committed:
Added:
- classifications_upload.py: handles the page to upload and classify an image
- classification_upload.html: html page for selecting the image

Modified:
- base.html : addded link to the new page

anauroranon committed:
Added comments

ManuMassi committed:
Added:
- requirements-macos.txt

ManuMassi committed:
Merge remote-tracking branch 'origin/Download_results_buttons' into Download_results_buttons

anauroranon committed:
Added comments

idduperezza committed:
added some comments

anauroranon committed:
Merge pull request #1 from ManuMassi/Color_Jitter_Transformation

Color jitter transformation

anauroranon committed:
Fixed file not found error

anauroranon committed:
Merge pull request #2 from ManuMassi/Color_Jitter_Transformation

Fixed file not found error

idduperezza committed:
Merge branch 'master' into Image_Histogram

idduperezza committed:
Merge pull request #3 from ManuMassi/Image_Histogram

Implemented image histogram

ManuMassi committed:
Merge branch 'master' into Download_results_buttons

ManuMassi committed:
Merge pull request #4 from ManuMassi/Download_results_buttons

Added download results buttons

anauroranon committed:
Merge branch 'master' into Upload_Image_Button

anauroranon committed:
Merge pull request #5 from ManuMassi/Upload_Image_Button

Upload image button

anauroranon committed:
bugfix relative to the name of the uploaded image

:)

anauroranon committed:
another bug fix

:D 💯

ManuMassi committed:
Create python-package.yml

anauroranon committed:
Delete .github/workflows directory

maurapintor commented on app/routes/classifications_jitter.py:
```diff
@@ -0,0 +1,86 @@
+@app.route('/classifications_jitter', methods=['GET', 'POST'])
+def classificationsJitter():
+    """
+    API for selecting a model and an image and running a
```
Please, update the docstring (the Jitter transformation should be mentioned) :book:

maurapintor commented on app/routes/classifications_upload.py:
```diff
@@ -0,0 +1,55 @@
+@app.route('/classifications_upload', methods=['GET', 'POST'])
+def classificationsUpload():
+    """
+    API for selecting an image from your computer
```
Check docstring here as well :smile:

maurapintor reviewed (changes requested):
Good job! There is just some docstring that needs attention, the rest looks good! :+1:

ManuMassi committed:
Improved docstring.
Modified classifications_upload.py: the file elimination happens now in a try except block to avoid server crashes.

maurapintor reviewed (approved):
Good job! :tada:
